libcaf_core/caf/detail/type_traits.hpp

```diff
@@ -980,6 +963,48 @@ class accepts_opaque_value {
   static constexpr bool value = sfinae_result::value;
 };
 
+/// Checks whether `T` is primitive, i.e., either an arithmetic type or
+/// convertible to one of STL's string types.
+template <class T, bool IsLoading>
+struct is_builtin_inspector_type {
+  static constexpr bool value = std::is_arithmetic<T>::value;
+};
+
+template <bool IsLoading>
+struct is_builtin_inspector_type<byte, IsLoading> {
+  static constexpr bool value = true;
+};
+
+template <bool IsLoading>
+struct is_builtin_inspector_type<span<byte>, IsLoading> {
+  static constexpr bool value = true;
+};
+
```

Comment:
Maybe I don't understand the question. A `span<const byte>` ist a range over immutable bytes. You can't load into that, because the bytes aren't writable.

Reply:
I think I'm not entirely sure what these do then and what the purpose of `IsLoading` is. What would this `is_builtin_inspector_type<span<byte>, true>::value` tell me?

Reply:
That `Inspector::is_loading` is `true`, i.e., you're dealing with a `load_inspector`. And only *save* inspectors accept immutable values like `string_view`.

Reply:
But how do you load into a `span<bytes>`, does this assume the span points to a byte region that is large enough to write into?

Reply:
What do you mean by "large enough"? A span has a fixed size. A byte span basically allows you to pass a fixed-size chunk of memory / BLOB to CAF without having CAF to know what the bytes actually represent.